### DISCUSSION TITLE
Preload songs before a previous one ends / Use songs from YouTube Music (songs, not music videos)

### DISCUSSION THREAD

**Discussion by zchrxh** [KRTirtho/spotube#856] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

There are some artists that have songs that seamlessly transition into each other where the song is preloaded by the official Spotify, Apple Music, YouTube Music clients, etc. Spotube doesn't do this and it would be really pleasing to see this.

Also, Spotube currently pulls some songs from their YouTube music video, which has non-song segments. SponsorBlock is used to fix this currently. But sometimes there are points in music videos that SponsorBlock doesn't pick up.

### Suggested solution

- Preload songs about 15-30 seconds before the previous song ends (or just preload the next song once the current song fully loads in).
- Pull songs from YouTube Music audios to prevent non-song segments. Like how in the NewPipe app, the user can select their search to only show the audio versions of songs through YouTube Music.

**Comment by KRTirtho** [comment]
We're already preloading the next song only after 20% progress of current track.

> Pull songs from YouTube Music audios to prevent non-song segments. Like how in the NewPipe app, the user can select their search to only show the audio versions of songs through YouTube Music.

Use Piped API with YouTubeMusic enabled by setting "Settings" > "API Type"  to "Piped" and "Settings" > "Search Mode" to "YouTube Music"

> Remember, Piped API can be broken sometime
> Use the `drgns.space` Piped instance which works a bit better

**Reply by ksharizard** [reply]
Spotube may preload the next song but it does not seamlessly transition into the next song like the other platforms do. There is always a 4-5 seconds delay before the next song is played which is annoying.

**Reply by KRTirtho** [reply]
You've to give a second to buffer. It doesn't support gapless playback
